sdcm/tester.py

```diff
@@ -1399,13 +1392,24 @@ def clean_resources(self):
     def tearDown(self):
         InfoEvent('TEST_END')
         self.log.info('TearDown is starting...')
+
+        with self.subTest("Check for Error/Critical Events"):
+            test_failing_events = self.get_critical_events()
+            if test_failing_events:
+                self.log.error('Error/Critical events found:')
+                self.log.error(test_failing_events)
+                self.fail('Error/Critical events found (see test logs)')
```

Comment:
will this run tearDown again? can it be an infinite loop?
I prefer that we hack a test result and not run like this.

Reply:
tearDown is always called once, self.fail just raise an exception.
since it's using subTest, this exception is just noted.

I think it's better to use subTest, then our own logic for doing that. at leas in this case.

there's no risk of infinite loop here